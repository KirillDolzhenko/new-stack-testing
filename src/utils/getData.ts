import axios from "axios";

export default async function () {
  try {
    let response: {
      data: {
        id: string;
      }[];
    } = await axios.get("https://jsonplaceholder.typicode.com/users");

    return response.data.map((el) => Number(el.id)).reduce((acc) => 1 + acc, 0);
  } catch (error) {
    console.log(error);
  }
}
