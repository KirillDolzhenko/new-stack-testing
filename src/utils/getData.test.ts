import getData from "./getData";
import axios from "axios";

jest.mock("axios");

describe("getData", () => {
  let response: any;

  beforeAll(() => {
    response = {
      data: [{ id: "5" }, { id: "5" }, { id: "5" }],
    };
  });

  test("axios data mock", async () => {
    (axios.get as jest.Mock).mockReturnValue(response);

    const data = await getData();

    expect(axios.get).toBeCalledTimes(1);
    expect(data).toBe(3);
    expect(data).toMatchSnapshot();
  });
});
