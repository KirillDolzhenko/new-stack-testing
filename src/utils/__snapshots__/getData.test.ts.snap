// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData axios data mock 1`] = `3`;
